src/__tests__/ParserComponent.test.tsx

```diff
@@ -265,6 +265,60 @@ describe('Csv FileEvents processed correctly\n', () => {
   });
 });
 
+describe('should accept valid csv file name with unusual' +
+    ' chars in file name', () => {
+  const props = {
+    prompt: 'test: ',
+    fileType: FileType.csv,
+  };
+
+
+  const onChangeMock = jest.fn();
+  const comp: any = mount(<ParserComponent
+    {...props}
+    onChange={onChangeMock}
+  />);
+
+  // clear the on change event mock and the enzyme component
+  beforeEach(() => {
+    onChangeMock.mockClear();
+    comp.setState({data: []});
+  });
+
+  it('file name with \\', async () => {
+    const testfile: File = new File(
+        ['Date,SomeNum,SomeString\n' +
+        '04/12/1998,4,abcd\n' +
+        '06-01-1994,5,efg\n' +
+        'November 5 1997,1,hij\n' +
+        ''],
+        'test\\.csv',
+        {type: '.csv,text/csv'},
+    );
+    const fileEvent = {target: {files: [testfile]}};
+    await comp.instance().parse(fileEvent);
+    expect(comp.state('data').length).toEqual(3);
+    expect(onChangeMock).toHaveBeenCalledTimes(1);
```

Comment:
Because you're checking this at the end of both tests, you can put this in afterEach()